x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -527,14 +528,13 @@ public void checkPrivileges(
         final Role userRole = ((RBACAuthorizationInfo) authorizationInfo).getRole();
         logger.trace(
             () -> new ParameterizedMessage(
-                "Check whether role [{}] has privileges cluster=[{}] index=[{}] application=[{}]",
+                "Check whether role [{}] has privileges [{}]",
                 Strings.arrayToCommaDelimitedString(userRole.names()),
-                Arrays.toString(privilegesToCheck.cluster()),
-                Arrays.toString(privilegesToCheck.index()),
-                Arrays.toString(privilegesToCheck.application())
+                privilegesToCheck
             )
         );
 
+        final ActionListener<PrivilegesCheckResult> maybeCachingListener;
```

Comment:
I'd suggestion we rename this variable to `listener` and the current `listener` to something like `originalListener`. There is a tendency that people would just call `listener` in future since it is the norm. So I think renaming it can help.